.env.sample

```diff
@@ -106,21 +115,33 @@ DEBUG=cache,presenters,events,emails,mailer,utils,http,server,services
 # set, all domains are allowed by default when using Google OAuth to signin
 ALLOWED_DOMAINS=
 
-# For a complete Slack integration with search and posting to channels the 
+# For a complete Slack integration with search and posting to channels the
 # following configs are also needed, some more details
 # => https://wiki.generaloutline.com/share/be25efd1-b3ef-4450-b8e5-c4a4fc11e02a
 #
 SLACK_VERIFICATION_TOKEN=your_token
 SLACK_APP_ID=A0XXXXXXX
 SLACK_MESSAGE_ACTIONS=true
 
-# Optionally enable google analytics to track pageviews in the knowledge base 
+# Display name for OIDC authentication
+OIDC_DISPLAY_NAME=OAuth2
+
+# Space separated auth scopes.
+OIDC_SCOPES=openid profile email
+
+# Override the claim used to assign users' usernames.
+OIDC_USERNAME_CLAIM=preferred_username
+
+# Override the claim used to assign the name of the team.
+OIDC_TEAM_CLAIM=team_name
```

Comment:
Could we move these up to L87 alongside the other `OIDC_`?

`OIDC_USERNAME_CLAIM` and `OIDC_TEAM_CLAIM` are unused in this PR, did you want to include the logic for that or remove these?

Reply:
> `OIDC_USERNAME_CLAIM` and `OIDC_TEAM_CLAIM` are unused in this PR, did you want to include the logic for that or remove these?

This is a good catch. I'm not actually pulling `username` from profile at all as the `accountProvisioner` is not expecting it. I have removed it.

On the other hand I did want to include `OIDC_TEAM_NAME` because it is not a standard claim, so other users will likely want to configure it to be whatever claim their IdP puts out.

Reply:
And can we get some example `OIDC_TEAM_NAME` values for the popular OIDC providers like Auth0/Keycloak/MS/etc? It’s definitely not standard and it would be nice to know which fields would “count” as a team name and how that relates to the authorization decisions :)

Reply:
Alternatively, because this authentication provider is only for use by self hosted installations where they will be a single team we could hard code the team name to something which can be changed later￼￼ by the user.

That removes this complication from the config entirely￼

Reply:
@tommoor That sounds feasible. Would that be something one would change through the Outline interface?

@JaneJeon that would be a bit difficult as there are no standing examples of it. It depends on the precise requirements of individual users. For example, with keycloak, one is able to define mappers attached to scopes or directly to clients that modify the contents of tokens and user info. Every identity provider probably does this differently, if they support it at all.

Reply:
as I've commented elsewhere in the PR, I believe the "proper" method would be to check against `aud`/`issuer`, not "team name", which are *guaranteed* to be there if you're using OIDC

Reply:
https://openid.net/specs/openid-connect-core-1_0.html#IDTokenValidation

Reply:
And again, id tokens (`sub`, `aud`, `issuer`, etc) - which is what you use to verify the identity of the user in a standardized manner - are *only* available for OIDC.

Reply:
I disagree that the "proper" method would be to check against any of the existing claims because that would overload their purpose.

The fact is that "team name" refers to an attribute that is explicitly used by Outline and serves a specific purpose. Additionally we cannot predict the contents of userinfo documents provided by all the identity providers this functionality will be used with in the future; despite being standardized there are plenty inconsistencies between implementations. Therefore the solution is to either specify some other way; perhaps as a separate configuration variable entirely; or we must make it configurable as to where the information comes from.

Furthermore, I would also disagree with setting the default claim name for specifying the team name to something like `iss` or `aud` because, while that may be applicable for some, it may be a source of misconfiguration error for others. It should be possible to set `OIDC_TEAM_CLAIM=aud`, but if that is the behavior the user wants then they should make an explicit decision to do so.

Reply:
One thing to consider here is that perhaps there shouldn't be a default at all... possibly the best behavior would be to remove the default and add an error prompting the user to configure it.

Reply:
> Would that be something one would change through the Outline interface?

Yes, it's already configurable. Honestly 99% of the reason this field exists is because the project is designed to be multi-tenant for cloud purposes. But as I mentioned before this is just not applicable in self-hosted (in fact, it's against the license terms) so it's fine to hard code this to "Wiki" or similar.

Reply:
Aha. This is fair. Changes incoming.